OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -236,5 +274,27 @@ void Explode(World world)
 
 			args.Weapon.Impact(Target.FromPos(pos), args.SourceActor, args.DamageModifiers);
 		}
+
+		bool AnyValidTargetsInRadius(World world, WPos pos, WDist radius, Actor firedBy, bool checkTargetType)
```

Comment:
To elaborate a bit: I thought about adding `ValidBlockers` and `BlockingStances` to `IBlocksProjectiles` traits so it's not simply "blocker" or "not blocker", but can be categorized.
For example, that would allow disc grenades to be 
 - unblockable before touching the ground (except by terrain or firestorm),
 - blocked by all walls and laserfences (friend or foe) once bouncing
 - blocked by enemy structures and vehicles, but not friendlies
 - not blocked by enemy infantry (unless directly targeted).

This is just an example, the 'infantry' part may be debatable.

More flexible blocker trait logic might be useful for 3rd-party mods as well, and might be potentially re-usable for activated firestorm (to let only friendly cluster/chem missiles pass, for example).